google/cloud/bigquery/job.py

```diff
@@ -3360,6 +3361,12 @@ def to_dataframe(
 
                 ..versionadded:: 1.24.0
 
+            date_as_object (bool):
```

Comment:
(nit) Should be `Optional[bool]` for consistency. And while at it, we can also fix the same for the `create_bqstorage_client` parameter.

The same for the parameter added in `table.py`.

Reply:
Ok, I think i need to take a separate cleanup task to make use of `optional` keyword consistent as the same method has different ways to use of  `optional` keyword.

Reply:
If you want to do this holistically across the class/module/library, that's totally fine, we don't have to bloat this PR.